src/uppy-base/src/plugins/Provider.js

```diff
@@ -8,12 +8,12 @@ export default class Provider {
   constructor (opts) {
     this.opts = opts
     this.provider = opts.provider
-    this.id = this.provider
+    this.id = opts.id || this.provider
```

Comment:
The reason I resolved to this is because of the inconsistency in googledrive's plugin name between uppy and uppy-server. When authenticating uppy-server expects http://localhost:3020/connect/google not http://localhost:3020/connect/drive. And this link is generated as seen [here](https://github.com/transloadit/uppy/pull/139/files#diff-535f9a41ac2aa69d7eb94840620d6eccR268). So I renamed the provider for google [here](https://github.com/transloadit/uppy/pull/139/files#diff-2179d365083ea9f487ccc5d64dba341fR24). But I don't know, there might be a better approach to this.